### DISCUSSION TITLE
Caught plotting error: Could not open /plot-

### DISCUSSION THREAD

**Discussion by Teledyol** [Chia-Network/chia-blockchain#1401] [Q&A]
Been trying with several versions for a few weeks and always stuck with plotting nothing

Here's the latest log (public keys truncated):

09:44:31.605  src.plotting.create_plots        : [32mINFO    [0m Creating 1 plots of size 32, pool public key:  959aed…
farmer public key: 8c6d7fc…
09:44:31.615  src.plotting.create_plots        : [32mINFO    [0m Memo: 959aed…[0m
09:44:31.615  src.plotting.create_plots        : [32mINFO    [0m Starting plot 1/1[0m

Starting plotting progress into temporary dirs: / and /
ID: 9eb98…
Plot size is: 32
Buffer size is: 4608MiB
Using 128 buckets
Using 2 threads of stripe size 65536
Caught plotting error: Could not open /plot-k32-2021-03-18-09-44-….plot.sort.tmp: Read-only file system
Traceback (most recent call last):
  File "src/cmds/chia.py", line 63, in <module>
  File "src/cmds/chia.py", line 59, in main
  File "click/core.py", line 1026, in __call__
  File "click/core.py", line 956, in main
  File "click/core.py", line 1518, in invoke
  File "click/core.py", line 1518, in invoke
  File "click/core.py", line 1280, in invoke
  File "click/core.py", line 711, in invoke
  File "click/decorators.py", line 22, in new_func
  File "src/cmds/plots.py", line 134, in create_cmd
  File "src/plotting/create_plots.py", line 164, in create_plots
RuntimeError: std::exception
[6290] Failed to execute script chia

**Comment by Teledyol** [comment]
Apologies…

Runiing macOS 11.3 on a Mac Pro (Late 2013) • 3.5 GHz 6-Core Intel Xeon E5 • MD FirePro D300 2 GB • APPLE SSD SM0256F

**Comment by hoffmang9** [comment]
1 one 3 things is going on.

1. You are using command line arguments around RAM and buckets that will not succeed
2. You have a problem with your disk
3. You have a RAM problem

3 is pretty remote on a Mac. What options are you using?

**Reply by Teledyol** [reply]
I'm assuming that plot space (101.4GiB) is too much. May be impossible unless I get larger SSDs

Not plotting parallel, even though I tried that for five days straight

8 plotes to an empty 2T disk

**Reply by Teledyol** [reply]
Using the GUI, not command line

**Comment by hoffmang9** [comment]
`Caught plotting error: Could not open /plot-k32-2021-03-18-09-44-….plot.sort.tmp: Read-only file system`
You don't have permissions to write a plot into the directories you are trying to.

**Reply by Teledyol** [reply]
That doesn't make sense. I'm showing that the disk is writable and that I'm the system admin.

I'm trying again in parallel, activity monitor is showing that each of the eight plots is opening at over 100% CPU and then down to above 60% for each plot. O it may have decided to work.

I shut down and ran harware tests and everything checks out on the internal system and the external farming disk

**Reply by Teledyol** [reply]
I'm assuming it's running parallel as long as %CPU stays above zero

Activity monitor is showing:

Process name • %CPU • CPU Time • threads etc.	
 chia	29.1	3:38.16	4	24	0.0	0.00	1381	
	
 chia	27.3	3:04.12	4	24	0.0	0.00	1389	
	
 chia	23.0	6:15.95	4	24	0.0	0.00	1345

 chia	22.2	1:01.48	4	17	0.0	0.00	1440
	
 chia	21.8	2:23.07	4	13	0.0	0.00	1401

 chia	21.8	1:23.45	4	18	0.0	0.00	1420

 chia	21.6	4:49.39	4	21	0.0	0.00	1369
	
 chia	21.6	1:53.72	4	17	0.0	0.00	1412	
 
 Chia Helper (Renderer)	10.3	1:58.85	26	3	0.0	0.00	995

**Reply by Teledyol** [reply]
That went on for about 8 - 3 minutes and is now showing zero

[debug.log](https://github.com/Chia-Network/chia-blockchain/files/6176545/debug.log)

**Reply by Teledyol** [reply]
maybe I should go back to "Add plot to Queue"

**Reply by Teledyol** [reply]
OmniDisk is showing the .chia folder taking up 92.8 GB and mainnet wrote a shit ton of plots from 0-251.7 MB before it stopped

I'm assuming that the SSD space is not adequate and that the 600 MiB setting might work better

**Reply by Teledyol** [reply]
Add to that log:

16:34:30.402 full_node full_node_server        : ERROR    Exception: unable to open database file <class 'sqlite3.OperationalError'>, closing connection {'host': '108.171.210.90', 'port': 8444}. Traceback (most recent call last):
  File "src/server/server.py", line 514, in api_call
  File "asyncio/tasks.py", line 481, in wait_for
  File "src/server/server.py", line 512, in wrapped_coroutine
  File "src/server/server.py", line 505, in wrapped_coroutine
  File "src/full_node/full_node_api.py", line 94, in new_peak
  File "src/full_node/full_node.py", line 343, in new_peak
  File "src/full_node/full_node.py", line 293, in short_sync_backtrack
  File "src/full_node/full_node.py", line 290, in short_sync_backtrack
  File "src/full_node/full_node.py", line 967, in respond_block
  File "src/consensus/blockchain.py", line 234, in receive_block
  File "src/full_node/block_store.py", line 75, in add_full_block
  File "aiosqlite/core.py", line 184, in execute
  File "aiosqlite/core.py", line 129, in _execute
  File "aiosqlite/core.py", line 102, in run
sqlite3.OperationalError: unable to open database file

16:34:30.429 full_node full_node_server        : WARNING  Banning 108.171.210.90 for 10 seconds
16:41:20.716 full_node full_node_server        : ERROR    Exception: unable to open database file, {'host': '92.33.205.219', 'port': 8444}. Traceback (most recent call last):
  File "src/server/server.py", line 505, in wrapped_coroutine
  File "src/full_node/full_node_api.py", line 94, in new_peak
  File "src/full_node/full_node.py", line 343, in new_peak
  File "src/full_node/full_node.py", line 293, in short_sync_backtrack
  File "src/full_node/full_node.py", line 290, in short_sync_backtrack
  File "src/full_node/full_node.py", line 967, in respond_block
  File "src/consensus/blockchain.py", line 234, in receive_block
  File "src/full_node/block_store.py", line 75, in add_full_block
  File "aiosqlite/core.py", line 184, in execute
  File "aiosqlite/core.py", line 129, in _execute
  File "aiosqlite/core.py", line 102, in run
sqlite3.OperationalError: unable to open database file

16:41:20.901 full_node full_node_server        : WARNING  Banning 92.33.205.219 for 10 seconds
16:41:21.180 full_node full_node_server        : WARNING  Banning 87.140.52.94 for 10 seconds
16:41:21.310 full_node full_node_server        : WARNING  Banning 23.242.220.172 for 10 seconds
16:41:21.694 full_node full_node_server        : WARNING  Banning 141.0.147.60 for 10 seconds
16:41:21.923 full_node full_node_server        : WARNING  Banning 146.168.248.216 for 10 seconds
16:41:22.171 full_node full_node_server        : WARNING  Banning 47.41.125.177 for 10 seconds

**Reply by Teledyol** [reply]
Is this IP getting banned?

**Comment by zachlankton** [comment]
I noticed from the first post your plot log has: 
`Starting plotting progress into temporary dirs: / and /`
You may be the system admin on your mac, but I wonder if the / root file system requires root user to write to?  Either way, you probably don't want to use your root file system as your temporary space.   **While testing a plot or 2 on the same hard drive as your operating system is fine, long term this is not recommended as it will wear out the disk and may cause loss of data or access to your computer.**

Your debug log is full of 
`wallet wallet_server              : ERROR    Exception: database or disk is full`
Do you have enough free disk space?

Additionally, 2TB is not enough for 8 parallel plots.  Remember that plotting requires more disk space than the final plot file.  This is covered in [plotting basics](https://www.chia.net/2021/02/22/plotting-basics.html).  [Also Here](https://github.com/Chia-Network/chia-blockchain/wiki/Beginners-Guide)

My best guess is right now you have a lot of temporary files left over from all of your previous plotting attempts that have filled your hard drive.

**Reply by Teledyol** [reply]
I guess one solution that I'm looking at is switching out to a 2.0TB NVMe SSD on the pro and writing 2-3 plots in parallel at a time.

Eventually, we're looking at the SuperMicro solution

**Reply by Teledyol** [reply]
Went with the 2TB SSD upgrade to my Mac Pro trashcan. A 32T raid0 for the Farm. Started plotting 4 in parallel and appears to be going well…

Had time to make a LorineChia icon for my farm disk
![Lorine_Chia](https://user-images.githubusercontent.com/3718370/112358936-0316c280-8c8e-11eb-899d-59e0dab5d25f.png)
